tikv/mvcc.go

```diff
@@ -1306,9 +1306,10 @@ func (store *MVCCStore) BatchGet(reqCtx *requestCtx, keys [][]byte, version uint
 	return pairs
 }
 
-func (store *MVCCStore) collectRangeLock(startTS uint64, startKey, endKey []byte, resolved []uint64) []*kvrpcpb.KvPair {
+func (store *MVCCStore) collectRangeLock(startTS uint64, startKey, endKey []byte, resolved []uint64, limit int) []*kvrpcpb.KvPair {
 	var pairs []*kvrpcpb.KvPair
 	it := store.lockStore.NewIterator()
+	var cnt int
 	for it.Seek(startKey); it.Valid(); it.Next() {
```

Comment:
Should we handle ReverseScan at here? It seems like this procedure always scan locks ascending, so it probably doesn't work properly in reverse mode now.
If we add a count limit here, we must handle reverse scan explicitly, because if we simply swap the `startKey` and `endKey` when invoke this funtion, it will skip some keys which we actually need it.

Reply:
Seems the `lockstore` iterator does not support reverse scan by now, maybe we could scan the whole range locks and only keep the valid pair collection limit check.

Reply:
We can use `Iterator.Prev` for ReverseScan. Because the `startKey`is the largest key bound in ReverseScan mode.

Reply:
But use `Prev` is a bit complicated, scan the whole range seems better.

Reply:
Got it.